network/dnstruncate/packet_proxy.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2023 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dnstruncate
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"sync/atomic"
+
+	"github.com/Jigsaw-Code/outline-internal-sdk/internal/slicepool"
+	"github.com/Jigsaw-Code/outline-internal-sdk/network"
+)
+
+// From [RFC 1035], the DNS message header contains the following fields:
+//
+//		                              1  1  1  1  1  1
+//		0  1  2  3  4  5  6  7  8  9  0  1  2  3  4  5
+//
+//	 +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
+//	 |                      ID                       |
+//	 +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
+//	 |QR|   Opcode  |AA|TC|RD|RA|   Z    |   RCODE   |
+//	 +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
+//	 |                    QDCOUNT                    |
+//	 +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
+//	 |                    ANCOUNT                    |
+//	 +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
+//	 |                    NSCOUNT                    |
+//	 +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
+//	 |                    ARCOUNT                    |
+//	 +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
+//
+// [RFC 1035]: https://datatracker.ietf.org/doc/html/rfc1035#section-4.1.1
+const (
+	dnsServerPort   = 53  // https://datatracker.ietf.org/doc/html/rfc1035#section-4.2
```

Comment:
This would be clearer:
```suggestion
	standardDNSPort   = 53  // https://datatracker.ietf.org/doc/html/rfc1035#section-4.2
```